test/terrain/notifications_test.clj

```diff
@@ -0,0 +1,32 @@
+(ns terrain.notifications-test
+  (:require [cemerick.url :as curl]
+            [clj-http.fake :refer [with-fake-routes-in-isolation]]
+            [clojure.java.io :refer [resource]]
+            [clojure.test :refer :all]
+            [medley.core :refer [map-vals]]
+            [terrain.clients.notifications :as nc]
+            [terrain.test-fixtures :as test-fixtures]
+            [terrain.util.config :as config]
+            [terrain.util.transformers :refer [add-current-user-to-map]]))
+
+(use-fixtures :once test-fixtures/with-test-config test-fixtures/with-test-user)
+
+(defn- notification-url [& components]
+  (str (apply curl/url (config/notificationagent-base-url) components)))
+
+(deftest last-ten-messages
+  (with-fake-routes-in-isolation
+    {{:address      (notification-url "messages")
+      :query-params (map-vals str (add-current-user-to-map nc/last-ten-messages-params))}
```

Comment:
`clj-http-fake` doesn't match query parameters with values that aren't strings, so we coerce the values in the map to be strings. I considered creating a custom function for this, but `map-vals` does exactly what we need relatively simply. Instead of adding a custom function, I'm planning to add a blurb to the contribution guidelines.